diff --git a/node_modules/boring-vault-ui/dist/contexts/v1/BoringVaultContextV1.js b/node_modules/boring-vault-ui/dist/contexts/v1/BoringVaultContextV1.js
index 3e3f49d..801ef9d 100644
--- a/node_modules/boring-vault-ui/dist/contexts/v1/BoringVaultContextV1.js
+++ b/node_modules/boring-vault-ui/dist/contexts/v1/BoringVaultContextV1.js
@@ -790,19 +790,19 @@ var BoringVaultV1Provider = function (_a) {
                 case 1:
                     _d.trys.push([1, 10, , 11]);
                     vaultContractWithSigner = new ethers_1.Contract(vaultContract, BoringVaultABI_1.default, signer);
-                    _a = Number;
+                    _a = BigInt;
                     _c = (_b = vaultContractWithSigner).allowance;
                     return [4 /*yield*/, signer.getAddress()];
                 case 2: return [4 /*yield*/, _c.apply(_b, [_d.sent(), withdrawQueueContract])];
                 case 3:
                     allowance = _a.apply(void 0, [_d.sent()]);
-                    if (!(allowance < amountWithdrawBaseDenom.toNumber())) return [3 /*break*/, 6];
+                    if (!(allowance < BigInt(amountWithdrawBaseDenom.toFixed(0)))) return [3 /*break*/, 6];
                     setWithdrawStatus({
                         initiated: true,
                         loading: true,
                     });
                     console.log("Approving token ...");
-                    return [4 /*yield*/, vaultContractWithSigner.approve(withdrawQueueContract, amountWithdrawBaseDenom.toNumber())];
+                    return [4 /*yield*/, vaultContractWithSigner.approve(withdrawQueueContract, amountWithdrawBaseDenom.toFixed(0))];
                 case 4:
                     approveTx = _d.sent();
                     return [4 /*yield*/, approveTx.wait()];
@@ -837,7 +837,7 @@ var BoringVaultV1Provider = function (_a) {
                         [
                             deadline.toNumber(), // Deadline
                             Number(0), //discountedSharePrice.toNumber(), // atomicPrice, this is actually overriden in safeUpdateAtomicRequest
-                            amountWithdrawBaseDenom.toNumber(), // offerAmount
+                            amountWithdrawBaseDenom.toFixed(0), // offerAmount
                             false, // inSolver
                         ], accountantContract, // accountant
                         formattedDiscountPercent.toNumber())];
